File: 45주차/올바른괄호.js
function solution(s) {
  var answer = true;
  var cnt = 0;
  for (let v of s) {
    if (v === ")") {
      cnt -= 1;
      if (cnt < 0) return false;
    } else cnt += 1;
  }

  return cnt === 0;
}
// 프로그래머스 Js로 풀기 - 스택. 큐

File: 47주차/올바른괄호.cpp
#include<string>
#include <iostream>

using namespace std;

bool solution(string s)
{
    bool answer = true;
    int cnt = 0;
    for (int i =0;i < s.size(); i++){
        if (s[i] == ')'){
            cnt -= 1;
            if (cnt < 0) return false;
        }
        else {
            cnt += 1;
        }
    }

    return cnt==0;
}